forward/ft_headmodel_duneuro.m

```diff
@@ -0,0 +1,243 @@
+if(isfield(duneuro_settings, 'type')) type = duneuro_settings.type;
+else type = 'fitted';
+end
+
+if(isfield(duneuro_settings, 'solver_type')) solver_type = duneuro_settings.solver_type;
+else solver_type = 'cg';
+end
+
+if(isfield(duneuro_settings, 'electrodes')) electrodes = duneuro_settings.electrodes;
+else electrodes = 'closest_subentity_center';
+end
+
+if(isfield(duneuro_settings, 'subentities')) subentities = duneuro_settings.subentities;
+else subentities = '1 2 3';
+end
+
+if(isfield(duneuro_settings, 'forward')) forward = duneuro_settings.forward;
+else forward = 'venant';
+end
+
+if(isfield(duneuro_settings, 'intorderadd')) intorderadd = duneuro_settings.intorderadd;
+else intorderadd = '2';
+end
+
+if(isfield(duneuro_settings, 'intorderadd_lb')) intorderadd_lb = duneuro_settings.intorderadd_lb;
+else intorderadd_lb = '2';
+end
+
+if(isfield(duneuro_settings, 'initialization')) initialization = duneuro_settings.initialization;
+else initialization = 'closest_vertex';
+end
+
+if(isfield(duneuro_settings, 'numberOfMoments')) numberOfMoments = duneuro_settings.numberOfMoments;
+else numberOfMoments = '3';
+end
+
+if(isfield(duneuro_settings, 'referenceLength')) referenceLength = duneuro_settings.referenceLength;
+else referenceLength = '20';
+end
+
+if(isfield(duneuro_settings, 'relaxationFactor')) relaxationFactor = duneuro_settings.relaxationFactor;
+else relaxationFactor = '1e-6';
+end
+
+if(isfield(duneuro_settings, 'restrict')) restrict = duneuro_settings.restrict;
+else restrict = 'true';
+end
+
+if(isfield(duneuro_settings, 'weightingExponent')) weightingExponent = duneuro_settings.weightingExponent;
+else weightingExponent = '1';
+end
+
+if(isfield(duneuro_settings, 'post_process')) post_process = duneuro_settings.post_process;
+else post_process = 'true';
+end
+
+if(isfield(duneuro_settings, 'subtract_mean')) subtract_mean = duneuro_settings.subtract_mean;
+else subtract_mean = 'true';
+end
+
+if(isfield(duneuro_settings, 'reduction')) reduction = duneuro_settings.reduction;
+else reduction = '1e-15';
+end
+
+
+if(isfield(duneuro_settings, 'intorderadd_meg')) intorderadd_meg = duneuro_settings.intorderadd_meg;
+else intorderadd_meg = '2';
+end
+
+if(isfield(duneuro_settings, 'mixedMoments')) mixedMoments = duneuro_settings.mixedMoments;
+else mixedMoments = 'true';
+end
+
+if(isfield(duneuro_settings, 'meg_type')) reduction = duneuro_settings.meg_type;
+else meg_type = 'physical';
+end
+
+if(isfield(duneuro_settings, 'meg_eneablecache')) meg_eneablecache = duneuro_settings.meg_eneablecache;
+else meg_eneablecache = 'false';
+end
+
+% start with an empty volume conductor
+mesh = ft_datatype_parcellation(mesh);
+headmodel = [];
+if isfield(mesh,'pos')
+  headmodel.pos = mesh.pos;
+else
+  error('Vertex field is required!')
+end
+
+if isfield(mesh,'tet')
+  headmodel.tet = mesh.tet;
+elseif isfield(mesh,'hex')
+  headmodel.hex = mesh.hex;
+  %%assume fieldtrip ordering, reorder to conform to DUNE standard
```

Comment:
should there be an explicit check on the ordering? I don't think that there is any specific "fieldtrip" ordering, or if there is, it is not explicitly defined.